test-framework/src/main/java/org/apache/brooklyn/test/framework/TargetableTestComponentImpl.java

```diff
@@ -116,4 +125,14 @@ private static Entity getTargetById(final ExecutionContext executionContext, fin
     protected <T> T getRequiredConfig(ConfigKey<T> key) {
         return checkNotNull(config().get(key), "config %s must not be null", key);
     }
+
+    protected void setUpAndRunState(boolean up, Lifecycle status) {
+        if (up) {
+            sensors().set(SERVICE_UP, up);
+            setExpectedState(this, status);
+        } else {
+            setExpectedState(this, status);
+            sensors().set(SERVICE_UP, up);
+        }
```

Comment:
We are waiting for `service.isUp` in `SoftwareProcess` so makes sense to do the same here for consistency.